db/migrate/20201119144047_drop_agents_table.rb

```diff
@@ -0,0 +1,8 @@
+class DropAgentsTable < ActiveRecord::Migration[6.0]
+  def up
+    drop_table :agents
+  end
+  def down
+    raise ActiveRecord::IrreversibleMigration
```

Comment:
just curiosity - I get the irreversible migration is there since we are dropping a table forever.. what are the benefits of raising an error to prevent the rollback of this? 
As in, isn't the action of dropping a table always not going to be rolled back? Just wondering if there are some rails mysteries you could enlighten me on :D

Reply:
There's a reasonable assumption that rolling back a migration may be destructive on the data within the database, so we don't need to `raise` to warn of that. But in this situation, the migration wouldn't even know how to rollback the scheme, because it doesn't know what structure of the `agent`s table was when it existed.

You can actually make a `drop_table` reversible, by giving it the state of the table when you dropped it, like:

```
drop_table :agents do |t|
  t.string :name
end
```

In which case the migration would be reversible. But we're unlikely to need to be able to do that, so I guess Rhian though this would suffice, but the `raise` makes it clear that this work hasn't been done, which I'm fine with.